### PR TITLE
Simple 

### PR DESCRIPTION
**Pull request by msaxena2**
WIP. Doesn't use Cell Fragments yet. Some issues with I/O need to be fixed.

### CODE REVIEW

**Comment by msaxena2** [comment]
Simple needs #1798 (or something that fixes #1682) to work.

File: k-distribution/tutorial/2_languages/1_simple/1_untyped/simple-untyped.k [hunk 1]
```diff
@@ -619,22 +636,22 @@ irrelevant once the function returns).  We add an artificial
 \texttt{nothing} value to the language, which is returned by the
 nulary \texttt{return;} statements. */
 
-  syntax KItem ::=  (Map,K,ControlCellFragment)
```
Comment (daejunpark) [inline k-distribution/tutorial/2_languages/1_simple/1_untyped/simple-untyped.k:622]: @msaxena2 You may want to refer to KOOL (untyped) language regarding the cell fragment, although SIMPLE has only a few uses of it.

File: k-distribution/tutorial/2_languages/1_simple/1_untyped/simple-untyped.k [hunk 2]
```diff
@@ -552,7 +560,8 @@ of the two argument values (which is what the equality on $K$ terms does). */
 /*@ The logical negation is clear, but the logical conjunction and disjunction
 are short-circuited: */
 
-  rule ! T => notBool(T)
```
Comment (daejunpark) [inline k-distribution/tutorial/2_languages/1_simple/1_untyped/simple-untyped.k:555]: @msaxena2 You may want to check this again on the latest master. I've had the same problem, but it was fixed at some point.

File: k-distribution/tutorial/2_languages/1_simple/1_untyped/simple-untyped.k [hunk 3]
```diff
@@ -1044,9 +1062,9 @@ The following operation initializes a sequence of locations with the same
 value: */
 
   syntax Map ::= Int "..." Int "|->" K
-    [function, latex({#1}\ldots{#2}\mapsto{#3})]
-  rule N...M |-> _ => .Map  when N >Int M
-  rule N...M |-> K => N |-> K (N +Int 1)...M |-> K  when N <=Int M
+    [function]
+  rule N ...M |-> _ => .Map  when N >Int M
```
Comment (daejunpark) [inline k-distribution/tutorial/2_languages/1_simple/1_untyped/simple-untyped.k]: @msaxena2 Do you know what is the problem here? How the space makes difference? Indeed, I have the same problem in KOOL language, but I didn't figure it out.

File: k-distribution/tutorial/2_languages/1_simple/1_untyped/simple-untyped.k [hunk 4]
```diff
@@ -153,9 +159,11 @@ Moreover, we want them to be strict, that is, to evaluate to lists of results
 whenever requested (e.g., when they appear as strict arguments of
 the constructs above). */
 
-  syntax Ids  ::= List{Id,","}
-  syntax Exps ::= List{Exp,","}          [strict]  // automatically hybrid now
-
+  syntax Bottom
```
Comment (daejunpark) [inline k-distribution/tutorial/2_languages/1_simple/1_untyped/simple-untyped.k:162]: @msaxena2 Could you please explain what is it for?

File: k-distribution/tutorial/2_languages/1_simple/1_untyped/simple-untyped.k (hunk 1 above)
Comment (msaxena2) [inline k-distribution/tutorial/2_languages/1_simple/1_untyped/simple-untyped.k:622]: Cell Fragments were broken when I was making these changes, so I found a way around them. But you're right about the usage of cell fragments in SIMPLE being limited. We can consider going back to cell fragments, given that #1745 fixed cell fragments.

File: k-distribution/tutorial/2_languages/1_simple/1_untyped/simple-untyped.k (hunk 4 above)
Comment (msaxena2) [inline k-distribution/tutorial/2_languages/1_simple/1_untyped/simple-untyped.k:162]: Kompile complains that the empty comma separated list has multiple sorts : Ids, Exps, and Vals. Making BottomList the least sort for a comma separated list solves that issue.

Comment (grosu) [inline k-distribution/tutorial/2_languages/1_simple/1_untyped/simple-untyped.k:162]: Of course, this is unacceptable.  Users should not have to do that.  That should be part of the translation from full K to KORE and not something users have to do themselves.

Comment (cos) [inline k-distribution/tutorial/2_languages/1_simple/1_untyped/simple-untyped.k:162]: @msaxena2, which part of Kompile complains?

**Comment by cos** [comment]
@msaxena2, we should me this forward. Did you get a chance to try the suggestions you received?

File: k-distribution/tutorial/2_languages/1_simple/1_untyped/simple-untyped.k (hunk 4 above)
Comment (msaxena2) [inline k-distribution/tutorial/2_languages/1_simple/1_untyped/simple-untyped.k:162]: @cos Sorry, it's not Kompile, but KRun that complains. The message I get, when I try running even very basic programs:

```
[Error] Critical: Cannot compute least sort of term: .List{"'_,_"}(.KList)
Possible least sorts are: [Exps, Ids, Vals]
All terms must have a unique least sort; consider assigning unique KLabels to
overloaded productions
```

**Comment by msaxena2** [comment]
@daejunpark @cos I've made some more changes incorporating your suggestions.

File: k-distribution/tutorial/2_languages/1_simple/1_untyped/simple-untyped.k (hunk 4 above)
Comment (cos) [inline k-distribution/tutorial/2_languages/1_simple/1_untyped/simple-untyped.k:162]: @daejunpark, did you encounter this problem with kool?

**Comment by cos** [comment]
@msaxena2, could you also add the ktests so we can merge?

**Comment by cos** [comment]
I understand some of the tests you've ignored depend on #1866. Did you figure out by any chance why the others are failing?

**Comment by msaxena2** [comment]
None of the failling tests depend on #1866 .

**Comment by cos** [comment]
> Did you figure out by any chance why the others are failing?

**Comment by msaxena2** [comment]
Jenkins: Test this please

**Comment by cos** [comment]
@daejunpark, thanks for agreeing to pick this up.

**Comment by daejunpark** [comment]
close via #1920

**Pull request closed without merging**